docs/source/entities.mdx

```diff
@@ -0,0 +1,321 @@
+
+Recall our example `Product` entity from the Products subgraph:
+
+```graphql
+# Products subgraph
+type Product @key(fields: "id") {
+  id: ID!
+  name: String!
+  price: Int
+}
+```
+
+With this definition in place, the Reviews subgraph can then add a field of type `Product` to its `Review` type, like so:
+
+```graphql{3}
+# Reviews subgraph
+type Review {
+  product: Product!
+  score: Int!
+}
+
+# This is a required "stub" of the Product entity (see below)
+type Product {
+  id: ID!
+}
+```
+
+To reference the `Product` entity, the Reviews subgraph needs to define a **stub** of it to make its own schema valid. The stub _doesn't_ need to include all `Product` fields. It _only_ needs to include `id`, because that's the entity's `@key` field. Without this field, the gateway doesn't know how to fetch the correct `Product` for a given `Review`.
+
+> Notice that the stub doesn't even need a `@key` directive! That's because the Reviews subgraph isn't adding any of its _own_ fields to `Product`. We'll see that later in [Extending entities](#extending-entities).
+
+This syntax has several benefits:
+
+* It's standard GraphQL grammar.
+* It enables you to run the `reviews` subgraph standalone with a valid schema, including a `Product` type with a single `id` field.
+* It provides strong typing information that lets you catch mistakes at schema composition time.
+
+## Resolving entities
+
+Let's say our Reviews subgraph from [Referencing an entity](#referencing-an-entity) defines the following `Query` type:
+
+```graphql:title=reviews
+type Query {
+  latestReviews: [Review!]
+}
+```
+
+That means the following query is valid against our federated graph:
+
+```graphql{6}
+query GetReviewsAndProducts {
+  latestReviews { # Defined in Reviews
+    score
+    product {
+      id
+      price # Not defined in Reviews!
+    }
+  }
+}
+```
+
+Now we have a problem: this query starts its execution in the Reviews subgraph (where `latestReviews` is defined), but that subgraph _doesn't_ know that `Product` entities have a `price` field! Remember, the Reviews subgraph only knows about the `id` field of `Product`.
+
+Because of this, the gateway needs to fetch `price` from the Products subgraph instead. But how does the gateway know _which_ products it needs to fetch the prices for?
+
+To solve this, we add a resolver to _each_ subgraph:
+
+* A resolver in Reviews to generate [representations](#entity-representations) of `Product` entities
+* A [reference resolver](#reference-resolvers) in Products to return full `Product` objects from representations
+
+### Entity representations
+
+In our example, the Reviews subgraph needs to define a resolver for its stub version of the `Product` entity. The Reviews subgraph doesn't know much about `Product`s, but fortunately, it doesn't need to. All it needs to do is return data for the fields it _does_ know about, like so:
+
+```js{3-10}:title=resolvers.js
+// Reviews subgraph
+const resolvers = {
```

Comment:
Something's wrong with that example: it's resolver for a `product` field in `Review` having one argument, but `Review.product` does not have arguments, and it references `review.productID` which isn't a field that has been defined anywhere.

What you need in the Reviews subgraph is an implementation for the `Query.latestReviews` field that has been hinted at earlier.

Reply:
I believe this is correct but requires some extra explanation. The parameter passed to this resolver isn't a GraphQL argument, but rather the `Review` object that was resolved just prior to resolving `Review.product`. That returned `Review` object can include a field named `productID`, pulled from the same data store that populated any actual supported fields of `Review`

Reply:
(Will indicate as much in the doc!)

Reply:
Ok, doc has been revamped pretty substantially and this example is no longer present.